### PR TITLE
[fixes #4262] add preprocessTree hooks to addons.

### LINKED ISSUE #4262: add Addon.prototype.preprocessTree
@chadhietala we should likely wait until the builder/pre-packager stuff lands right?

### PR DESCRIPTION
**Pull request by stefanpenner**
- [ ] tests

### CODE REVIEW

**Comment by rwjblue** [comment]
Seems good to me.

**Comment by stefanpenner** [comment]
Ya. I have tests locally. Will push them later tonight

**Comment by trabus** [comment]
I love hooks. :+1:

File: lib/broccoli/ember-app.js [hunk 1]
```diff
@@ -366,6 +390,52 @@ EmberApp.prototype.addonPostprocessTree = function(type, tree) {
...
 };
 
+
+/**
+  Runs addon postprocessing on a given tree and returns the processed tree.
+
+  This enables addons to do process immediately **before** the preprocessor for a
+  given type is run, but before concatenation occurs.  If an addon wishes to
+  apply a transform  after the preprocessors run, they can instead implement the
+  postprocessTree hoo.
```
Comment (Robdel12) [inline lib/broccoli/ember-app.js]: > postprocessTree hoo**k**.

**Pull request closed without merging**